Mathlib/LinearAlgebra/TensorProduct/Pi.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Judith Ludwig, Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Christian Merten
+-/
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+import Mathlib.LinearAlgebra.Pi
+
+/-!
+
+# Tensor product and products
+
+In this file we examine the behaviour of the tensor product with arbitrary and finite products.
+
+Let `S` be an `R`-algebra, `N` an `S`-module and `ι` an index type. We then have a natural map
+
+`TensorProduct.piScalarRightHom`: `N ⊗[R] (ι → R) →ₗ[S] (ι → N)`
+
+In general, this is not an isomorphism, but if `ι` is finite, then it is
+and it is packaged as `TensorProduct.piScalarRight`.
+
+-/
+
+variable (R : Type*) [CommRing R]
+variable (S : Type*) [CommRing S] [Algebra R S]
+variable (N : Type*) [AddCommMonoid N] [Module R N] [Module S N] [IsScalarTower R S N]
+variable (ι : Type*)
+
+open LinearMap TensorProduct
+
+private def TensorProduct.piScalarRightHomBil : N →ₗ[S] (ι → R) →ₗ[R] (ι → N) where
```

Comment:
You can use `namespace TensorProduct` since a lot of declarations start like that.